### ISSUE TITLE
🐞[BUG]: State plugin not having state after ofActionComplete(UpdateState)

### ISSUE THREAD

**Issue by lurock** [ngxs/store#1857]
### Affected Package

 The issue is caused by package @ngxs/storage-plugin

### Is this a regression?

 Yes, the previous version in which this bug was not present was: 3.7.3

### Description

  Selecting state using this.store.selectSnapshot method after ofActionCompleted(UpdateState) on a browser refresh when using the Storage plugin worked fine in version 3.7.3 but in 3.7.4 it stopped working.  The state loaded from the storage plugin is no longer available at that time.

## Environment

<pre><code>
Libs:
- @angular/core version: 14.0.0
- @ngxs/store version: 4.7.4

**Comment by arturovt** [comment]
Hey @lurock . Thanks for the quick response on the release. Is there a possibility to provide a reproducible example?

**Comment by arturovt** [comment]
@lurock could you try installing the `3.7.4-dev.master-310a613` version and give it a try? Just update all packages to the `3.7.4-dev.master-310a613` version as follows:
```json
"@ngxs/store": "3.7.4-dev.master-310a613",
"@ngxs/storage-plugin": "3.7.4-dev.master-310a613"
```

**Comment by lurock** [comment]
@arturovt I tried it with the 3.7.4-dev.master-310a613 version and it still does not have the state restored when the ofActionComplete(UpdateState) subscription fires.   

I have setup a repository with an example of the issue with NGXS on version 3.7.4: https://github.com/lurock/ngxs-state-storage-restore

If you look at the code you will see I console.log out the state after UpdateState is completed on this line https://github.com/lurock/ngxs-state-storage-restore/blob/7c5b527fda7a8bd823c5219ce10d1830dbdee8e5/src/app/actions-listener.service.ts#L19

For contrast I have setup a working version in a separate branch.  The only difference is the version of NGXS is now on 3.73.
https://github.com/lurock/ngxs-state-storage-restore/tree/ngxs-working

Steps to reproduce:
git clone https://github.com/lurock/ngxs-state-storage-restore.git
yarn install
yarn start

Launch the app at http://localhost:4200
1. click the save button to change the state
2. look at the console output the state is not restored from the storage plugin
3. Ctrl-c to stop the app

To compare with working version:
git checkout ngxs-working
yarn install
yarn start

Launch the app at http://localhost:4200
1. click the save button to change the state
2. look at the console output the state is now restored from the storage plugin

**Comment by markwhitfeld** [comment]
Great news! [v3.7.5](https://github.com/ngxs/store/releases/tag/v3.7.5) has been released and it includes a fix for this issue.
We are closing this issue, but please re-open it if the issue is not resolved.
Please leave a comment in the [v3.7.5 release discussion](https://github.com/ngxs/store/discussions/1909) if you come across any regressions with respect to your issue.

**Comment by bthharper** [comment]
@markwhitfeld  - I am still getting this issue with 3.7.5 (well actually https://github.com/ngxs/store/issues/1877).  

```typescript
    NgxsStoragePluginModule.forRoot({
      key: ['auth.callbackNavigationUrl']
    }),
```

I have the above in a submodule, and the auth.callbackNavigationUrl is cleared on the @@UPDATE_STATE action.

Here's the console log where the `auth.callbackNavigationUrl` state argument is being cleared.

![image](https://user-images.githubusercontent.com/8340092/196563819-1a985d0c-8096-4183-aaa7-19df0b4cdc00.png)

I can raise a new issue but I wanted to check if I was making a mistake first.

**Comment by arturovt** [comment]
@bthharper I’ve fixed this issue in the last commit, that would be shipped with 3.7.6

**Comment by bthharper** [comment]
> @bthharper I’ve fixed this issue in the last commit, that would be shipped with 3.7.6

Thanks @arturovt - do you have any timescale for 3.7.6 being released? 

Can I use the latest dev version using @ngxs/store@dev?

**Comment by arturovt** [comment]
> Thanks @arturovt - do you have any timescale for 3.7.6 being released?

No unfortunately. Mark is responsible for releases.

> Can I use the latest dev version using @ngxs/store@dev?

Yes, it's 2 commits ahead of master, see https://github.com/ngxs/store/blob/master/CHANGELOG.md#to-become-next-patch-version.
